app/src/main/java/lt/gyvosistorijos/MainActivity.kt

```diff
@@ -1,77 +1,112 @@
 package lt.gyvosistorijos
 
+import android.content.res.Resources
 import android.os.Bundle
 import android.support.v7.app.AppCompatActivity
 import com.bluelinelabs.conductor.Conductor
 import com.bluelinelabs.conductor.Router
 import com.bluelinelabs.conductor.RouterTransaction
-import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.MapStyleOptions
 import kotlinx.android.synthetic.main.activity_main.*
 import lt.gyvosistorijos.location.GeofenceHelper
+import lt.gyvosistorijos.location.LocationService
 import lt.gyvosistorijos.manager.RemoteConfigManager
+import timber.log.Timber
+
 
 class MainActivity : AppCompatActivity() {
 
-    internal lateinit var router: Router
+    companion object {
+        val MAP_SAVED_STATE_KEY = "map"
+    }
 
-    internal lateinit var map: MapboxMap
+    internal var router: Router? = null
 
-    internal lateinit var geofenceHelper: GeofenceHelper
+    internal lateinit var map: GoogleMap
 
+    internal lateinit var geofenceHelper: GeofenceHelper
+    internal lateinit var locationService: LocationService
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        router = Conductor.attachRouter(this, controller_container, savedInstanceState)
-
-        mapView.onCreate(savedInstanceState)
-
         RemoteConfigManager.instance.fetchConfig()
-
         geofenceHelper = GeofenceHelper(this, RemoteConfigManager.instance)
+        locationService = LocationService(this)
+
+        mapView.onCreate(savedInstanceState?.getBundle(MAP_SAVED_STATE_KEY))
+
+        mapView.getMapAsync { googleMap ->
+            map = googleMap
 
-        mapView.getMapAsync { mapboxMap ->
-            map = mapboxMap
+            map.uiSettings.isMyLocationButtonEnabled = false
 
-            map.uiSettings.isTiltGesturesEnabled = false
-            map.uiSettings.isRotateGesturesEnabled = false
+            initMapStyle()
 
+            val router = Conductor.attachRouter(this, controller_container, savedInstanceState)
             if (!router.hasRootController()) {
                 router.setRoot(RouterTransaction.with(SyncController()))
```

Comment:
`MainController` currently assumes there's a map to use. So it's either this or propagate the map loaded event down via some mechanism.